apollo-federation/src/sources/connect/models.rs

```diff
@@ -74,6 +77,41 @@ impl ConnectorBatchSettings {
     }
 }
 
+#[derive(Debug, Clone, Default)]
+pub struct ConnectorErrorsSettings {
+    pub message: Option<JSONSelection>,
+    pub extensions: Option<JSONSelection>,
+}
+
+impl ConnectorErrorsSettings {
+    fn from_directive(
+        connect_errors: Option<&ErrorsArguments>,
+        source_errors: Option<&ErrorsArguments>,
+    ) -> Self {
+        // `errors` set at @connect always overrides whatever is set at @source
+        let message = if let Some(connect_errors) = connect_errors {
+            connect_errors.message.clone()
+        } else if let Some(source_errors) = source_errors {
+            source_errors.message.clone()
+        } else {
+            None
+        };
+
+        let extensions = if let Some(connect_errors) = connect_errors {
+            connect_errors.extensions.clone()
+        } else if let Some(source_errors) = source_errors {
+            source_errors.extensions.clone()
+        } else {
+            None
+        };
+
+        Self {
+            message,
+            extensions,
+        }
```

Comment:
```suggestion
        let message = connect_errors
            .and_then(|e| e.message.as_ref())
            .or_else(|| source_errors.and_then(|e| e.message.as_ref())).cloned();

        let extensions = connect_errors
            .and_then(|e| e.extensions.as_ref())
            .or_else(|| source_errors.and_then(|e| e.extensions.as_ref())).cloned();

        Self {
            message,
            extensions,
        }
```

i _think_ this is both shorter and fixes a logic bug. if my brain is working, what you have doesn't support this situation:

```
@source(errors: { extensions: "foo" })
@connect(errors: { message: "bar" })
```

Because `connect_errors` is `Some`, we'll try to grab `extensions` off it and get `None`, and we won't even check `source_errors.extensions`.